tests/lib/rules/no-new-func.js

```diff
@@ -21,7 +21,14 @@ const ruleTester = new RuleTester();
 ruleTester.run("no-new-func", rule, {
     valid: [
         "var a = new _function(\"b\", \"c\", \"return b+c\");",
-        "var a = _function(\"b\", \"c\", \"return b+c\");"
+        "var a = _function(\"b\", \"c\", \"return b+c\");",
+        {
+            code: "class Function {}; new Function()",
+            parserOptions: {
+                ecmaVersion: 2015
+            }
+        },
+        "function Function() {}; Function()"
```

Comment:
Can we add a test where `class Function {}` and `function Function() {}` are defined in an inner scope? I think it would be beneficial to add tests for the cases where they are defined in a class method, a block, and in a function body.